source/includes/connection-snippets/x509-snippets/go-connection.go

```diff
@@ -13,7 +13,7 @@ import (
 func main() {
 	ctx := context.TODO()
 	certificateKeyFilePath := "/etc/certs/mongodb/client.pem"
-	uri := "mongodb+srv://<cluster-url>/test?authSource=$external&tlsCertificateKeyFile=%s&retryWrites=true&w=majority&authMechanism=MONGODB-X509"
+	uri := "mongodb+srv://<cluster-url>/test?authSource=$external&tlsCertificateKeyFile=%s&retryWrites=true&w=majority&authMechanism=MONGODB-X509&tls=true"
```

Comment:
Since this is injecting an arbitrary string into the URI via `%s`, I think you'll want to ensure that the value is URL encoded (as discussed in the [connection string spec](https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst)). This also relates to [Section 3.4 Query of RFC 3986](https://tools.ietf.org/html/rfc3986#section-3.4)). If there is an alternative method to provide this option to the Client, that may be easier.

@mongomoe: note that this applies to connection strings for _all_ languages, not just Go.

Reply:
Second thought: perhaps we don't care about URL encoding because the path is defined on the line above and does not contain any special characters. Feel free to disregard this comment.